### PR TITLE
fix: uuid for option keys

### LINKED ISSUE #2188: fix(db-select): options by prop leads to duplicated keys
### Which generators are impacted?

- [ ] All
- [ ] HTML
- [X] React
- [ ] Angular
- [ ] Vue
- [ ] Web components
- [ ] Power Apps

### Reproduction case

when using `<DBSelect>` with props `option` the generated key isn't unique. This leeds to an error message in the developer  console:

```
Warning: Each child in a list should have a unique "key" prop.
```

### Expected Behaviour

Generated key should be unique

### Screenshots

_No response_

### Browser version

None

### Add any other context about the problem here.

_No response_

### PR DESCRIPTION
**Pull request by annsch**
## Proposed changes

fixes #2188

## Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: packages/components/src/components/select/select.lite.tsx [hunk 1]
```diff
@@ -128,17 +127,14 @@ export default function DBSelect(props: DBSelectProps) {
 							<>
 								<Show when={option.options}>
 									<optgroup
-										key={'optgroup-' + option.value}
+										key={'optgroup-' + uuid()}
```
Comment (nmerget) [inline packages/components/src/components/select/select.lite.tsx:130]: I would suggest that we pass an `id` to the option and use `uuid()` as fallback:

`model.ts
export type DBSelectOptionType = {
id?:string
}
`

Comment (annsch) [reply packages/components/src/components/select/select.lite.tsx:130]: @nmerget fixed this for options

File: packages/components/src/components/select/select.lite.tsx [hunk 2]
```diff
@@ -137,7 +136,9 @@ export default function DBSelect(props: DBSelectProps) {
 												<option
 													key={
 														'option-' +
-														optgroupOption.value
+														optgroupOption.id
+															? optgroupOption.id
+															: uuid()
```
Comment (mfranzke) [inline packages/components/src/components/select/select.lite.tsx:141]:
```suggestion
														optgroupOption.id || uuid()
```

**Comment by mfranzke** [comment]
@annsch is this _ready for review_?

**Comment by annsch** [comment]
> @annsch is this _ready for review_?

@mfranzke yes, please

File: packages/components/src/components/select/select.lite.tsx (hunk 2 above)
Comment (nmerget) [reply packages/components/src/components/select/select.lite.tsx]: > 

@annsch and I talked and we don't need a custom `id` on the `key={}`. `uuid()` is totally fine

**Review by nmerget — APPROVED** [review]

**Review by nmerget — APPROVED** [review]

**Pull request merged**